src/compiler/rust.rs

```diff
@@ -1844,6 +1844,81 @@ fn test_can_trim_this() {
     assert!(!can_trim_this(&rlib_file));
 }
 
+#[cfg(feature = "dist-client")]
+fn maybe_add_cargo_toml(input_path: &Path, verify: bool) -> Option<PathBuf> {
+    let lib_rs = PathBuf::new().join("src").join("lib.rs");
+    if input_path.ends_with(lib_rs) {
+        let cargo_toml_path = input_path
+            .parent()
+            .expect("No parent")
+            .parent()
+            .expect("No parent")
+            .join("Cargo.toml");
+        // We want to:
+        //  - either make sure the file exists (verify=true)
+        //  - just return the path (verify=false)
+        if cargo_toml_path.is_file() || !verify {
+            Some(cargo_toml_path)
+        } else {
+            None
+        }
+    } else {
+        None
+    }
+}
+
+#[test]
+#[cfg(feature = "dist-client")]
+fn test_maybe_add_cargo_toml() {
+    #[cfg(not(target_os = "windows"))]
+    {
+        let root = PathBuf::from("/home/user/mozilla-source/mozilla-unified/third_party/rust");
```

Comment:
Can we just create a `root` and `result_cargo` variables with [this feature](https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-macro), for example?

Looks like another test part may be equal for both Windows & non-Windows platforms, isn't it?

Reply:
i splitted the test per-platform mostly because `"/home/user/mozilla-source/mozilla-unified/third_party/rust/wgpu-core/Cargo.toml"` would not match when ran on windows.

I added the assert on `to_str()` result to verify the actual path, but maybe this is useless? What's your opinion? If we remove this we can remove the conditionals; I could also welcome feedback on whether it is useful we test with unix and windows path on all platforms?

Reply:
What about something like this in the beginning of the test?

```rust
let (root, result_cargo_toml_path) = if cfg!(windows) {
  (
    "/home/user/mozilla-source/mozilla-unified/third_party/rust",
    "/home/user/mozilla-source/mozilla-unified/third_party/rust/wgpu-core/Cargo.toml"
  )
} else {
  (
    "C:\\mozilla-source\\mozilla-unified\\third_party\\rust",
    "C:\\mozilla-source\\mozilla-unified\\third_party\\rust\\wgpu-core\\Cargo.toml"
  )
};

...<cross-platform-test-with-checks>...
```

P.S. yes, `\` symbols must be escaped inside the string literals in Rust too.

Reply:
@AJIOB that's not the problem I was facing, rather that resulting string on windows for the `.to_str()` assert was:
> "/home/user/mozilla-source/mozilla-unified/third_party/rust/wgpu-core\\Cargo.toml"

Reply:
It's almost impossible, if you have `"C:\\mozilla-source\\mozilla-unified\\third_party\\rust"` as a source string

Reply:
> It's almost impossible, if you have `"C:\\mozilla-source\\mozilla-unified\\third_party\\rust"` as a source string

that's my second question: does it makes sense to have the test running both variants on all paltforms?

Reply:
I think it should be Windows variant on Windows platform, Linux variant on Linux platform

Reply:
any reason why you prefer a raw string over a `PathBuf::from()` in your example?

Reply:
Then you can build `PathBuf::from()` from that raw strings.
I've just trying to minimize what we really need to be platform-dependent inside the tests

Reply:
Thanks, I wanted to make sure there was no other reason!